cupy/linalg/einsum.py

```diff
@@ -18,22 +18,32 @@ def calc_single_view(ioperand, subscript):
             more than once, calculate diagonal for those axes.
     """
 
-    assert ioperand.ndim == len(subscript)
+    if '@' in subscript:
```

Comment:
I think at most one `@` exist. If so, when `'@' in subscript` `subsceript.count('@') == 1`. Is it right?

Reply:
Ah, Yes.
I make assertion more simple. (l.22-l.23.)